### PR TITLE
add task execute feature

### PR DESCRIPTION
**Pull request by krish8484**
- following functions have been added 
dot_product
mat_add
mat_subract

### CODE REVIEW

File: scheduler.py [hunk 1]
```diff
@@ -14,28 +14,33 @@ def __init__(self):
...
         signal.signal(signal.SIGINT, self.sigterm_handler)
 
     def submit_task(self, task : Task) -> Future:
-        random_worker = random.choice(self.workers)
+        if self.globalIncrementalWorkerId > 1:
+            random_worker_id = random.randint(0, self.globalIncrementalWorkerId - 1)
+        else:
+            random_worker_id = 0
+        print(random_worker_id)
+        random_worker = self.workers[random_worker_id]
```
Comment (aadhithNarayan) [inline scheduler.py:26]: Any specific reason why we moved from random.choice to this?

File: scheduler.py [hunk 2]
```diff
@@ -14,28 +14,33 @@ def __init__(self):
...
         self.globalIncrementalWorkerId = 0; # A globally incrementing worker id maintained by the scheduler
         signal.signal(signal.SIGINT, self.sigterm_handler)
 
     def submit_task(self, task : Task) -> Future:
-        random_worker = random.choice(self.workers)
+        if self.globalIncrementalWorkerId > 1:
+            random_worker_id = random.randint(0, self.globalIncrementalWorkerId - 1)
+        else:
+            random_worker_id = 0
+        print(random_worker_id)
```
Comment (aadhithNarayan) [inline scheduler.py]: Can you please convert this to a logging stmt or remove if not needed?

File: worker.py [hunk 3]
```diff
@@ -47,3 +63,22 @@ def notify_task_completion(self, task):
     def sigterm_handler(self, signum, frame):
         logging.info("Exiting gracefully.")
         sys.exit(0)
+
+    def decode_argument(self, arg):
```
Comment (aadhithNarayan) [inline worker.py:67]: Can we move the operation methods, decode logic and operations mapping(dict) to a separate class/file (eg. operations.py) and consume it here? 
Note: This is a clean code suggestion. Could be taken up in subsequent PRs.

File: scheduler.py (hunk 1 above)
Comment (krish8484) [reply scheduler.py:26]: We have a worker id as a primary key now, we should use that instead of other contents

File: worker.py (hunk 3 above)
Comment (krish8484) [reply worker.py:67]: Thanks for the suggestion. Can take up if we have more functions

File: scheduler.py (hunk 2 above)
Comment (krish8484) [reply scheduler.py]: done

**Review by aadhithNarayan — APPROVED** [review]

**Review by swarnim-vyas — APPROVED** [review]

**Review by swarnim-vyas — COMMENTED** [review]
So now the task Id is also flowing correctly right?

**Pull request merged**